sdk/storage/Azure.Storage.Common/src/Shared/StorageBearerTokenChallengeAuthorizationPolicy.cs

```diff
@@ -49,21 +49,26 @@ public StorageBearerTokenChallengeAuthorizationPolicy(TokenCredential credential
 
         /// <inheritdoc />
         protected override void AuthorizeRequest(HttpMessage message)
-        {
-            if (tenantId != null || !_enableTenantDiscovery)
-            {
-                base.AuthorizeRequest(message);
-            }
-        }
+            => AuthorizeRequestInternal(message, false).EnsureCompleted();
 
         /// <inheritdoc />
         protected override ValueTask AuthorizeRequestAsync(HttpMessage message)
+            => AuthorizeRequestInternal(message, true);
+
+        private async ValueTask AuthorizeRequestInternal(HttpMessage message, bool async)
         {
             if (tenantId != null || !_enableTenantDiscovery)
             {
-                return base.AuthorizeRequestAsync(message);
+                TokenRequestContext context = new TokenRequestContext(_scopes, message.Request.ClientRequestId);
```

Comment:
We should send the tenantId along as well here. 

```suggestion
                TokenRequestContext context = new TokenRequestContext(_scopes, message.Request.ClientRequestId, tenantId: tenantId);
```